AAEmu.Game/Models/Game/Char/Character.cs

```diff
@@ -1364,9 +1378,15 @@ public override void SetPosition(float x, float y, float z, sbyte rotationX, sby
             var moved = !Position.X.Equals(x) || !Position.Y.Equals(y) || !Position.Z.Equals(z);
             var lastZoneKey = Position.ZoneId;
             base.SetPosition(x, y, z, rotationX, rotationY, rotationZ);
+            
+            if (Position.Z < 98 && !IsUnderWater) //TODO: Need way to determine when player is under any body of water. 
+                IsUnderWater = true;
+            else if (Position.Z > 98 && IsUnderWater)
+                IsUnderWater = false;
```

Comment:
Might want to move this under "if (!moved)"?

Also hacky as fuck but will kill anyone falling out of bounds which is 👍

Reply:
This is above moved because if the player dies and re-spawns on position and does not move the breath bar does not show until they move again.

Reply:
for performance sake I can switch the if statement around to isUnderwater is checked first.